src/main/scala/org/constellation/storage/TransactionService.scala

```diff
@@ -65,7 +69,13 @@ class TransactionService[F[_]: Sync](dao: DAO)
       .flatMap(_ => txs.map(tx => accepted.remove(tx.transaction.hash)).sequence.void)
   }
 
-  def pullForConsensus(minCount: Int): F[List[TransactionCacheData]] = {
+  def pullForConsensusSafe(minCount: Int, semaphore: Semaphore[F], roundId: String = "roundId")
+                          (implicit F: Concurrent[F]): F[List[TransactionCacheData]] = {
+    val lock = new SingleLock[F, List[TransactionCacheData]](roundId, semaphore, pullForConsensus(minCount, roundId))
+    lock.acquire
+  }
+
+  def pullForConsensus(minCount: Int, roundId: String = "roundId"): F[List[TransactionCacheData]] = {
```

Comment:
I see the as of yet unused `roundId` string here. Will this used for some filtering, debug or why is it here too?

Reply:
Just debugging for now. Still testing